lib/github-interface.js

```diff
@@ -0,0 +1,232 @@
+}
+
+// Returns a repo's tags in reverse chronological order
+async function getTags ({ repo }) {
+  const tags = await request({
+    method: 'GET',
+    url: `${githubApiBase}/repos/${repo}/tags`,
+    auth,
+    json: true,
+    headers: githubHeaders
+  })
+
+  return tags.map((tag) => tag.name)
+}
+
+// Adds the given version number as a tag to all of the supplied repos and a prerelease
+// with the supplied message to the first repo in the array.
+// Returns a promise that resolves to the Github API release-creation response
+//
+// For example:
+//
+// prerelease({
+//   repos: ['tableflip/project-frontend', 'tableflip/project-api'],
+//   version: 'v2.0.0',
+//   message: 'Adds notifications and basic admin console'
+// })
+async function prerelease ({ repos, version, message }) {
+  const tags = repos.map((repo) => tagRepo({ repo, version }))
+  await Promise.all(tags)
+  return prereleaseRepo({ repo: repos[0], version, message })
+}
+
+// Adds a the supplied version tag to the current master branch in the given repo
+async function tagRepo ({ repo, version }) {
+  const lastCommit = await request({
+    method: 'GET',
+    url: `${githubApiBase}/repos/${repo}/commits/master`,
+    auth,
+    json: true,
+    headers: githubHeaders
+  })
+
+  if (!lastCommit || !lastCommit.sha) throw new Error(`Cannot get latest sha for the repo ${repo}`)
+
+  return request({
+    method: 'POST',
+    url: `${githubApiBase}/repos/${repo}/git/refs`,
+    auth,
+    json: {
+      ref: `refs/tags/${version}`,
+      sha: lastCommit.sha
+    },
+    headers: githubHeaders
+  })
+}
+
+// Adds a release (with the prerelease flag) to the supplied version, with the supplied message, to the given repo
+function prereleaseRepo ({ repo, version, message }) {
+  return request({
+    method: 'POST',
+    url: `${githubApiBase}/repos/${repo}/releases`,
+    auth,
+    json: {
+      tag_name: version,
+      target_commitish: 'master',
+      name: version,
+      body: message,
+      prerelease: true
+    },
+    headers: githubHeaders
+  })
+}
+
+// Gets the package.json for a repo, assuming it's located in the root.  Throws if no such package.json is found
+async function getPackageJson ({ repo }) {
+  const res = await request({
+    method: 'GET',
+    url: `${githubApiBase}/repos/${repo}/contents/package.json`,
+    auth,
+    json: true,
+    headers: githubHeaders
+  })
+  const buf = Buffer.from(res.content, 'base64')
+  return {
+    json: JSON.parse(buf.toString('utf8')),
+    sha: res.sha
+  }
+}
+
+// Returns the version number from the given repo's package.json, or throws if it can't be found
+async function getVersion ({ repo }) {
+  const { json } = await getPackageJson({ repo })
+  return json.version
+}
+
+// Returns the supplied version number bumped for the next release.
+// Default releaseType is a 'patch', but could be 'major' or 'minor'.
+function bumpVersion ({ version, releaseType = 'patch' }) {
```

Comment:
I think this should be required and not default, we don't want to accidentally release a patch that was meant to be a major/minor.